### PR TITLE
PyPanda Expect: Allow to turn off unansi

### PR DESCRIPTION
**Pull request by pcworld**
Not all shells use ANSI, and also unansi is still incomplete and thus
may sometimes corrupt output.
Add serial_kwargs to Panda() constructor to avoid cluttering the
constructor with new arguments in the future if Expect adds new
arguments.
Further make sure not to use mutable default parameters in Expect.

### CODE REVIEW

File: panda/python/core/pandare/panda_expect.py [hunk 1]
```diff
@@ -386,8 +385,12 @@ def expect(self, expectation=None, timeout=30):
...
                 # note this drops the echo'd command
                 if self.current_line.endswith(b"\n"):
-                    # End of line - need to unansi and move into prior_lines
-                    self.unansi()
+                    # End of line - need to potentially unansi and move into prior_lines
+                    if self.use_unansi:
+                        self.unansi()
+                    else:
+                        self.prior_lines.append(self.current_line[:-1].decode(errors='ignore'))
+                        self.current_line = bytearray()
```
Comment (pcworld) [inline panda/python/core/pandare/panda_expect.py:393]: I'm not completely sure whether this is sufficient in all cases, so feel free to take a careful look.

**Comment by pcworld** [comment]
Also note that this disables unansi in `panda.monitor_console`. I assumed that qemu's monitor does not use ANSI codes when used over a socket, however I might just as well be wrong, in which case we would need to pass `unansi=True` there. (Or alternatively, make `unansi=True` the default in `Expect.__init__`.)

**Comment by AndrewFasano** [comment]
Thanks for helping clean this up! ANSI codes are a pain and I'm not proud of the code that handles them.

I'll do some testing of this before merging, I feel like the qemu monitor was using ANSI but I'll double check that.

And good catch on the mutable default parameter!

**Comment by AndrewFasano** [comment]
Looks like I was wrong, the QEMU monitor isn't doing any ANSI shenanigans. I do notice one difference though - but I think it's probably okay. Previously the lines of output from the qemu monitor were joined by `\n`s while now they're joined with `\r\n`s - do you think that's okay or should we find a way to strip the `\r`s?

Output from running `print(repr(panda.run_monitor_cmd("info registers")[:100]))`:

Old:
```
'EAX=00000020 EBX=c1d878a0 ECX=00000000 EDX=000003f8\nESI=f38b0000 EDI=c14e25b0 EBP=f5c35f38 ESP=f5c35'
```

New:
```
'EAX=00000103 EBX=c1d878a0 ECX=00000000 EDX=000003f8\r\nESI=f38b0000 EDI=0000000e EBP=f5c35f38 ESP=f5c3'
```

**Comment by pcworld** [comment]
> ANSI codes are a pain and I'm not proud of the code that handles them.

I mean, it's cool that it exists.
Though it took me a while to find out *that* it exists (also the magic that attempts to remove the command string). Which is why I've now documented this a bit in Panda's constructor.

> Previously the lines of output from the qemu monitor were joined by `\n`s while now they're joined with `\r\n`s - do you think that's okay or should we find a way to strip the `\r`s?

I think that's okay. If people used Python's [str.splitlines](https://docs.python.org/3/library/stdtypes.html#str.splitlines) for splitting the lines, this would be transparent anyway.
As far as I've understood you're now following a (versioned?) release model with stable releases. Maybe we should introduce some changelog that lists breaking changes such as this? (Would also be relevant for #1041, give me a few days to respond in that issue.)

Btw, I think that removing the command string and unansi'ing might be considered different things (and thus it might make sense to put these behind different options). Though separating these two features would be a bit more work, and I personally don't need it so far, so I would leave it as is in this PR.

**Comment by AndrewFasano** [comment]
Sorry, I imagine that was pretty confusing to track down! Sometimes we (foolishly) assume the features we need for working with our generic qcows are features everyone will always want.

We wrote up our plan for releases on the project wiki [here](https://github.com/panda-re/panda/wiki/PANDA-Branches-&-Versioning) - I think we're supposed to be including version numbers on the dev branch (whoops, we haven't been doing that) and building a changelog. I was thinking we'd just do that when we updated the stable branch. I'll add a changelog and version number in a few minutes and include an entry for this change.

I agree that removing the echo'd command and unansi'ing are different but I'm happy to leave that change for another time when someone actually needs it!

Thanks for all the recent PRs 😄!

**Comment by AndrewFasano** [comment]
Update: I think I was wrong about the qemu monitor not using ANSI commands- see this error here:
https://github.com/panda-re/panda/pull/806/checks?check_run_id=3185790820#step:2:283. That said, this is the only place I've seen ANSI output from the monitor and I tested it a few times. I'll investigate a bit more.

**Comment by AndrewFasano** [comment]
That link seems to have broken since I posted it- the monitor_cmds.py test failed once because it got what looks like an ANSI message back from the qemu monitor when it was expecting a string. I haven't been able to reproduce the error though and it's gone on subsequent runs of the CI.

**Pull request merged**